modules/SRanipal_module/src/EyelidsRetargeting.cpp

```diff
@@ -0,0 +1,244 @@
+{
+    m_desiredEyeOpennes = eyeOpennes;
+}
+
+bool EyelidsRetargeting::update()
+{
+    if (!m_configured)
+    {
+        yError() << "[EyelidsRetargeting::update] The eyelids retargeting is not configured.";
+        return false;
+    }
+
+    int eye_open_level = static_cast<int>(std::round(m_desiredEyeOpennes / m_eyeOpenPrecision));
+    double eye_openess_leveled = m_eyeOpenPrecision * eye_open_level;
+
+    if (m_useRawEyelids)
+    {
+        if (eye_open_level != m_eyeOpenLevel)
+        {
+            yarp::os::Bottle& out = m_rawEyelidsOutputPort.prepare();
+            out.clear();
+            double rawEyelidsValue
+                    = eye_openess_leveled * (static_cast<double>(m_rawEyelidsOpenValue) - m_rawEyelidsCloseValue)
+                    + m_rawEyelidsCloseValue;
+            out.addString("S" + std::to_string(static_cast<int>(std::round(rawEyelidsValue))));
+            m_rawEyelidsOutputPort.write();
+            m_eyeOpenLevel = eye_open_level;
+            yInfo() << "[EyelidsRetargeting::update] Setting eye openess (raw):" << eye_openess_leveled;
+            yDebug() << "[EyelidsRetargeting::update] Sending raw commands to eyelids:" << out.toString();
+        }
+    }
+    else
+    {
+        double eyelidsReference = (1.0 - eye_openess_leveled) * (m_maxEyeLid - m_minEyeLid) + m_minEyeLid; // because min-> open, max->closed
+        if (m_useEyelidsPositionControl)
+        {
+            if (eye_open_level != m_eyeOpenLevel)
+            {
+                if (m_eyelidsPos)
+                {
+                    m_eyelidsPos->positionMove(0, eyelidsReference); // because min-> open, max->closed
+                }
+                m_eyeOpenLevel = eye_open_level;
+                yInfo() << "[EyelidsRetargeting::update] Setting eye openess (position):" << eye_openess_leveled;
+            }
+        }
+        else
+        {
+            if (m_eyelidsVel && m_eyelidsEnc)
+            {
+                if (eye_open_level != m_eyeOpenLevel)
+                {
+                    m_eyeOpenLevel = eye_open_level;
+                    yInfo() << "[EyelidsRetargeting::update] Setting eye openess (velocity):" << eye_openess_leveled;
+                }
+
+                double currentEyelidPos = 0;
+                if (m_eyelidsEnc->getEncoder(0, &currentEyelidPos))
+                {
+                    double eyelidsVelocityReference = m_eyelidsVelocityGain * (eyelidsReference - currentEyelidPos);
+                    double eyelidsVelClamped = std::max(-m_eyelidsMaxVelocity, std::min(eyelidsVelocityReference, m_eyelidsMaxVelocity)); //Clamp the velocity reference in [-max, +max];
+                    m_eyelidsVel->velocityMove(0, eyelidsVelClamped);
+                }
+                else
+                {
+                    yWarning() << "[EyelidsRetargeting::update] Failed to get eyelids encoder value. Skipping control.";
+                    return false;
```

Comment:
It will not fail. See https://github.com/robotology/walking-teleoperation/blob/98e6ebd835627b48799979d21e8a9cb893b5ad16/modules/SRanipal_module/src/SRanipalModule.cpp#L177-L180

The return value is willingly ignored